src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,236 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.LayoutView;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * Instantiates and performs failure detection adn handling asynchronously.
+ * <p>
+ * Created by zlokhandwala on 9/28/16.
+ */
+@Slf4j
+public class ManagementServer extends AbstractServer {
+
+    @Override
+    public void reset() {
+    }
+
+    @Override
+    public void reboot() {
+    }
+
+    /**
+     * The options map.
+     */
+    private Map<String, Object> opts;
+    private ServerContext serverContext;
+
+    private IFailureDetectorPolicy failureDetectorPolicy;
```

Comment:
Allowing a race condition for "performance" purposes doesn't sound right. I was also not implying that they should be blocking calls. 

Also, counting failed pings since the last successful ping might not be a good metrics, it would be able to discover a node failure, but not a fault. A slow host can miss the right pings such the the counter is always reset and never passes the threshold.

You don't have to change your policy/metric code in this patch, but this is something for you to think about.